polars/polars-lazy/polars-plan/src/logical_plan/optimizer/cse_expr.rs

```diff
@@ -13,21 +16,21 @@ use crate::prelude::visitor::{ALogicalPlanNode, AexprNode, RewritingVisitor, Tre
 /// have little collisions
 /// We will do a full expression comparison to check if the
 /// expressions with equal identifiers are truly equal
-// TODO! try to use a hash `usize` for this?
-type Identifier = Rc<str>;
+type Identifier = u64;
```

Comment:
This also needs the left-most leaf column name to be able to determine the output. So `(u64, Arc<str>)`.